pkg/collector/reconcile/service.go

```diff
@@ -71,9 +71,8 @@ func desiredService(ctx context.Context, params Params) *corev1.Service {
 	labels := collector.Labels(params.Instance, []string{})
 	labels["app.kubernetes.io/name"] = naming.Service(params.Instance)
 
-	// by coincidence, the selector is the same as the label, but note that the selector points to the deployment
-	// whereas 'labels' refers to the service
-	selector := labels
+	selector := collector.SelectorLabels(params.Instance)
+	selector["app.kubernetes.io/name"] = naming.Service(params.Instance)
```

Comment:
Do we need this label? there is already `"app.kubernetes.io/instance":   naming.Truncate("%s.%s", 63, instance.Namespace, instance.Name),`